.travis.yml

```diff
@@ -94,6 +94,42 @@ matrix:
         # Report success since we have overridden default behaviour
         - bash -c "$STATUS" success "Local $NAME testing has passed"
 
+    - env:
+        - NAME=astyle
+      install:
+      - wget https://downloads.sourceforge.net/project/astyle/astyle/astyle%203.1/astyle_3.1_linux.tar.gz;
+        mkdir -p BUILD && tar xf astyle_3.1_linux.tar.gz -C BUILD;
+        pushd BUILD/astyle/build/gcc;
+        make;
+        export PATH=$PWD/bin:$PATH;
+        popd;
+      - astyle --version
+      script:
+        # only changed files to be checked for astyle
+        - git diff --name-only $TRAVIS_BRANCH | grep '.*\.\(h\|c\|hpp\|cpp\)' | fgrep -v -f .astyleignore | xargs -n 100 -I {} bash -c "astyle -n --options=.astylerc \"{}\"" > astyle.out;
```

Comment:
So master will always say "Passed, 0 warnings".

We could also run the astyle branch on the current commit without printing the warnings so the status says "Passed, 204 warnings" on master. On the PR we can make it say "Passed, 204 warnings (+3 warnings)".

Reply:
Is that necessary ? Master files are going to be cleaned up all of them (once this is approved, I'll start module by module). Thus master should be always 0. Only PR that modifies the files can be >0. Or?

Reply:
The problem is this code will say "Passed, 0 warnings" even if there are 100 warnings.

> More PR will come later to start updating the code base

So we don't have 0 warnings yet. Am I wrong?

If we merge this as is your future PRs aren't going to be very impressive. What will they say? "I removed 50 astyle warnings, reducing the total from 0 warnings to 0 warnings"? : )

Reply:
Still not completely certain about the solution but would it be:

1. we do astyle on master branch
2. we do astyle on changes (as it is now here)
3. we show these : master + PR astyle warnings (+/- only PR astyle warnings)

Is that correct?

Reply:
Oh! There status script will grab whatever number you have on master and automatically compare with that.


The only thing that needs to change is that CURR is set to the the total number of warnings.

So you probably just need to run astyle again without the changes.

Note we don't need to print out all of the warnings, the diff is more useful.